### PR TITLE
Moved cloning contributors to a function

### PR DESCRIPTION
**Pull request by sosdmike**
Delete the byline if there are contributors and add them to the content item.

### CODE REVIEW

File: p2p/__init__.py [hunk 1]
```diff
@@ -602,44 +602,46 @@ def clone_content_item(self, slug, clone_slug, keep_embeds=False, keep_relateds=
...
-            if len(clone_contributors) > 0:
-                try:
-                    self.append_contributors_to_content_item(clone['id'], clone_contributors)
-                except P2PException:
-                    raise
 
-            return clone['id']
+        return clone['id']
+
+    def get_cloned_contributors(self, content_item):
```
Comment (charlex) [inline p2p/__init__.py]: If we're only using this function in one place, why must it be a function?

File: p2p/__init__.py [hunk 2]
```diff
@@ -602,44 +602,46 @@ def clone_content_item(self, slug, clone_slug, keep_embeds=False, keep_relateds=
...
-                            # If a match was found, update the entry with the staff slug
-                            byline_item = {'slug': contributor['slug']}
-
-                    # Add the final result to the clone_contributors array
-                    clone_contributors.append(byline_item);
-
-            # If we have some contributors, let's add them
-            if len(clone_contributors) > 0:
-                try:
-                    self.append_contributors_to_content_item(clone['id'], clone_contributors)
```
Comment (charlex) [inline p2p/__init__.py]: Can we remove the `append_contributors_to_content_item()` method? Is this just cruft now?

Comment (jperezlatimes) [inline p2p/__init__.py]: It's still an active endpoint in CS

**Comment by jperezlatimes** [comment]
Ok, once we can get a hold of Pete we'll merge this in.

File: p2p/__init__.py (hunk 2 above)
Comment (charlex) [inline p2p/__init__.py]: We should slate it for deletion when SNAP no longer uses it. No need to keep unused cruft now that CS has consolidated the contributor endpoints into the content_item endpoint. 

Be gone bloat, be gone.

Comment (sosdmike) [inline p2p/__init__.py]: We can raise a DeprecationWarning for this function and remove in another version.

File: p2p/__init__.py (hunk 1 above)
Comment (sosdmike) [inline p2p/__init__.py]: It doesn't have to be a function but I like compartmentalizing what is happening instead of inlining it.

Comment (sosdmike) [inline p2p/__init__.py]: Function calls are fairly cheap and keep the code easier to read IMHO.

Comment (charlex) [inline p2p/__init__.py]: Let's at least throw an underscore on the front of it and make it a private method. As someone who uses dir() to debug and figure out what the heck I can use in code, this kind of thing cornfuses me.

Mind throwing a docstring on it as well?

```
"""
Parses a content item payload's freeform bylines and advanced bylines
into a XXX
"""
```

^ Mind helping me write this?

File: p2p/__init__.py (hunk 2 above)
Comment (charlex) [inline p2p/__init__.py]: Sounds good.

**Comment by charlex** [comment]
Anyone ran the test suite on this yet?

File: p2p/__init__.py (hunk 1 above)
Comment (sosdmike) [inline p2p/__init__.py]: Sure, lets do that.

File: p2p/__init__.py (hunk 2 above)
Comment (sosdmike) [inline p2p/__init__.py]: @charlex Added a warning on the most recent commit.

Comment (charlex) [inline p2p/__init__.py]: Great improvement @sosdmike. I dig the warnings file.

Comment (sosdmike) [inline p2p/__init__.py]: just an FYI you can see the warnings if you run python with `python -wd <script>`

**Comment by sosdmike** [comment]
@charlex All the tests pass on the most recent commit.

**Comment by charlex** [comment]
Thank you for running them Mike. My SSL config on my machine is busted, so I'm having difficulty running them.

File: p2p/__init__.py [hunk 3]
```diff
@@ -602,44 +603,52 @@ def clone_content_item(self, slug, clone_slug, keep_embeds=False, keep_relateds=
                 }
                 payload['related_items'].append(related_item)
 
+        contributors = self._get_cloned_contributors(content_item)
+
+        if contributors:
+            del content_item['byline']
+            content_item['contributors'] = contributors
```
Comment (sosdmike) [inline p2p/__init__.py]: Yep, let me fix that up.

**Comment by jperezlatimes** [comment]
Merging in

**Pull request merged**